daemon/config/config_test.go

```diff
@@ -15,38 +15,42 @@ import (
 	"github.com/spf13/pflag"
 	"gotest.tools/v3/assert"
 	is "gotest.tools/v3/assert/cmp"
-	"gotest.tools/v3/fs"
 	"gotest.tools/v3/skip"
 )
 
+func makeConfigFile(t *testing.T, content []byte) string {
+	t.Helper()
+
+	name := filepath.Join(t.TempDir(), "daemon.json")
+	f, err := os.Open(name)
```

Comment:
Should probably be `os.Create()` here, as the file won't exist, but instead of this, just `os.Writefile()` would probably make things a lot easier;

```go
func makeConfigFile(t *testing.T, content []byte) string {
	t.Helper()
	name := filepath.Join(t.TempDir(), "daemon.json")
	err := os.WriteFile(name, content, 0o666)
	assert.NilError(t, err)
	return name
}
```

We can consider using a `string` for the content, as that's what used in most cases, which makes the utility slightly easier to use;

```go
func makeConfigFile(t *testing.T, content string) string {
	t.Helper()
	name := filepath.Join(t.TempDir(), "daemon.json")
	err := os.WriteFile(name, []byte(content), 0o666)
	assert.NilError(t, err)
	return name
}
```

Reply:
Fair point on `WriteFile` being cleaner here. However, I do want to keep `content` as `[]byte`; I have a follow-up PR to stack on top of this one that will make use of the byte array/that will require changing it back.

If you'd like, we can add a wrapper that accepts a UTF-8 string and converts to a bytestring?

Reply:
Ah, it looks like I'm misremembering Go string semantics; they are just bytestrings conventionally but not normatively UTF-8. Evidently I've spent too much time in Rust and Python lately.

I think we can go ahead and just make this strings then; my other PR should rebase on top just fine.